lib/model/model.go

```diff
@@ -1320,25 +1501,44 @@ func (m *model) handleAutoAccepts(deviceCfg config.DeviceConfiguration, folder p
 
 			fcfg := config.NewFolderConfiguration(m.id, folder.ID, folder.Label, fs.FilesystemTypeBasic, filepath.Join(defaultPath, path))
 			fcfg.Devices = append(fcfg.Devices, config.FolderDeviceConfiguration{
-				DeviceID: deviceCfg.DeviceID,
+				DeviceID: deviceID,
 			})
 
-			l.Infof("Auto-accepted %s folder %s at path %s", deviceCfg.DeviceID, folder.Description(), fcfg.Path)
+			if len(ccDevice.EncPwToken) > 0 || len(ccDeviceUs.EncPwToken) > 0 {
+				fcfg.Type = config.FolderTypeReceiveEncrypted
+			}
```

Comment:
The token is always on the device that is encrypted. I.e. when the remote generates its cluster config, if it is a receive-encrypted folder (i.e. remote is untrusted), the remote token is not empty, otherwise if it encrypts data for us, the token for our device is not empty. That's also why it's impossible for both the token for us and the remote to be non-empty: That would mean the remote itself has encrypted data/is untrusted, and it actively encrypts data sent to us - not possible.

Reply:
Why can't I have:

`<untrusted> <-> <untrusted>` but both know/use the password, effectively adding an extra encryption layer on top of the transport?

In that case you'd have tokens from both sides, and that would be fine, as long as they match.

Reply:
I assume you mean the situation as in the first graphic here: https://docs.syncthing.net/branch/untrusted/html/users/untrusted.html  
That is possible, however both won't send a token for themselves, only for the remote. Because they themselves do not have encrypted data, they do encrypt outgoing data. I.e. for this scenario in this condition `ccDevice` (now `ccDeviceRemote`) will be empty.